### PR TITLE
Jline

### PR DESCRIPTION
**Pull request by eatkins**
- [x] I've read the [CONTRIBUTING](https://github.com/sbt/sbt/blob/develop/CONTRIBUTING.md) guidelines

### CODE REVIEW

File: main/src/main/scala/sbt/Defaults.scala [hunk 1]
```diff
@@ -264,15 +264,6 @@ object Defaults extends BuildCommon {
           Clean.deleteContents(tempDirectory, _ => false)
       },
       useSuperShell :== sbt.internal.TaskProgress.isEnabled,
-      progressReports := { (s: State) =>
```
Comment (eed3si9n) [inline main/src/main/scala/sbt/Defaults.scala]: I consider progressReports to be somewhat stable actually.
super shell is one implementation on it.

File: internal/util-complete/src/main/scala/sbt/internal/util/LineReader.scala [hunk 2]
```diff
@@ -120,7 +122,7 @@ private[sbt] object JLine {
   protected[this] val originalIn = new FileInputStream(FileDescriptor.in)
 
   private[sbt] def makeInputStream(injectThreadSleep: Boolean): InputStream =
-    if (injectThreadSleep) new InputStreamWrapper(originalIn, Duration("50 ms"))
+    if (injectThreadSleep) new InputStreamWrapper(originalIn, 2.milliseconds)
```
Comment (eed3si9n) [inline internal/util-complete/src/main/scala/sbt/internal/util/LineReader.scala:125]: 👍

File: main/src/main/scala/sbt/internal/TaskProgress.scala [hunk 3]
```diff
@@ -25,7 +25,7 @@ private[sbt] final class TaskProgress
   private[this] val isStopped = new AtomicBoolean(false)
 
   override def initial(): Unit = {
-    ConsoleAppender.setTerminalWidth(JLine.usingTerminal(_.getWidth))
+    ConsoleAppender.setTerminalWidth(JLine.terminal.getWidth)
```
Comment (eed3si9n) [inline main/src/main/scala/sbt/internal/TaskProgress.scala:28]: 👍

File: main/src/main/scala/sbt/Defaults.scala (hunk 1 above)
Comment (eed3si9n) [reply main/src/main/scala/sbt/Defaults.scala]: One thing I might change about the `progressReports` is maybe dropping `s: State` and just making it a Seq of TaskProgress.

Comment (eatkins) [reply main/src/main/scala/sbt/Defaults.scala]: Aah, ok. That makes sense. I understand what the key is better now than when I first wrote the commit. It was in fact the State parameter that was especially throwing me off.

**Review by eed3si9n — CHANGES_REQUESTED** [review]
I am happy with the first two commits, so could you split the last one out?

**Comment by eatkins** [comment]
Yep. Been working on some other stuff but will re-push once I get to this.

**Review by eed3si9n — APPROVED** [review]

**Pull request merged**